pkg/controllers/management/multiclusterapp/controller_multiclusterapp.go

```diff
@@ -418,18 +468,24 @@ func getAnswerMap(answerMap map[string]map[string]string, projectName string) ma
 }
 
 // deleteApps finds all apps created by this multiclusterapp and deletes them
-func (m *MCAppController) deleteApps(mcAppName string, mcapp *v3.MultiClusterApp) (runtime.Object, error) {
+func (m *MCAppManager) deleteApps(mcAppName string, mcapp *v3.MultiClusterApp) (runtime.Object, error) {
 	// get all apps with label "multiClusterAppId" = name of this app
 	appsToDelete := []*pv3.App{}
-	set := labels.Set(map[string]string{multiClusterAppIDSelector: mcAppName})
+	projectToCluster := map[string]string{}
+	clusterContexts := map[string]*config.UserContext{}
+	set := labels.Set(map[string]string{MultiClusterAppIDSelector: mcAppName})
 	var err error
 
 	if mcapp == nil {
-		appsToDelete, err = m.getAllApps(mcAppName, true)
+		appsToDelete, projectToCluster, clusterContexts, err = m.getAllAppsWithContexts(mcAppName)
 		if err != nil {
 			return nil, err
 		}
 	} else {
+		clusterContexts, err = m.getClusterContexts(mcapp)
```

Comment:
we should consider moving ns creation to the user controller

Reply:
opened issue to keep track `https://github.com/rancher/rancher/issues/18189`